src/models/booking.cr

```diff
@@ -1,6 +1,49 @@
+require "json"
+
+struct History
+  include JSON::Serializable
+
+  property state : Booking::State
+  property time : Int64
+  property source : String?
+
+  def initialize(@state : Booking::State, @time : Int64, @source : String? = nil)
+  end
+end
+
+class HistoryConverter
+  def self.to_column(x) : Array(History)?
+    case x
+    when Nil
+      nil
+    when JSON::Any
+      Array(History).from_json x.to_json
+    else
+      raise "Cannot convert from #{x.class} to Array(History)"
+    end
+  end
+
+  def self.to_db(x : Array(History)?)
+    x.to_json
+  end
+end
+
```

Comment:
These should probably go into a separate file.
I would follow the class hierarchy like so...

```crystal
# In `booking/history.cr`
struct History
  include JSON::Serializable

  property state : Booking::State
  property time : Int64
  property source : String?

  def initialize(@state : Booking::State, @time : Int64, @source : String? = nil)
  end
  
  class Converter
    def self.to_column(x) : Array(History)?
      case x
      when Nil
        nil
      when JSON::Any
        Array(History).from_json x.to_json
      else
        raise "Cannot convert from #{x.class} to Array(History)"
      end
    end
  
    def self.to_db(x : Array(History)?)
      x.to_json
    end
  end
end
```

Reply:
But try requiring at the top of the file for experiment's sake

Reply:
how about doing it this way?
```cr
class Booking
  struct History
    ...
  end
end
```

Reply:
That works too. I would try to make sure that it's closer to the bottom of the class so it doesn't distract from the important components of the data model, i.e. the columns of `Booking`

Reply:
The way you have it now is good, so I'll resolve this.

Reply:
I mean, still move it to "booking/history.cr" but nest it instead of doing "struct Booking::History" and then having to move the require to the bottom of the file.

Reply:
Ah, in that case... the same error applies, unfortunately.

Reply:
You have to do the same require workaround regardless

Reply:
```cr
# booking/history.cr

class Booking
  struct History
    ...
  end
end
```

```cr
# booking.cr

require "./booking/history"

class Booking
  ...
end
```

The require can be at the top if doing this.